internal/services/mongocluster/mongo_cluster_resource.go

```diff
@@ -0,0 +1,572 @@
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r MongoClusterResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MongoCluster.MongoClustersClient
+
+			id, err := mongoclusters.ParseMongoClusterID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r MongoClusterResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var state MongoClusterResourceModel
+			if err := metadata.DecodeDiff(&state); err != nil {
+				return fmt.Errorf("DecodeDiff: %+v", err)
+			}
+
+			switch state.CreateMode {
+			case string(mongoclusters.CreateModeDefault):
+				if state.AdministratorUserName == "" {
+					return fmt.Errorf("`administrator_username` is required when `create_mode` is %s", string(mongoclusters.CreateModeDefault))
+				}
+
+				if state.AdministratorPassword == "" {
+					return fmt.Errorf("`administrator_password` is required when `create_mode` is %s", string(mongoclusters.CreateModeDefault))
+				}
+
+				if state.ComputeTier == "" {
+					return fmt.Errorf("`compute_tier` is required when `create_mode` is %s", string(mongoclusters.CreateModeDefault))
+				}
+
+				if state.StorageSizeInGb == 0 {
+					return fmt.Errorf("`storage_size_in_gb` is required when `create_mode` is %s", string(mongoclusters.CreateModeDefault))
+				}
+
+				if state.HighAvailabilityMode == "" {
+					return fmt.Errorf("`high_availability_mode` is required when `create_mode` is %s", string(mongoclusters.CreateModeDefault))
+				}
+
+				if state.ShardCount == 0 {
+					return fmt.Errorf("`shard_count` is required when `create_mode` is %s", string(mongoclusters.CreateModeDefault))
+				}
+
+				if state.Version == "" {
+					return fmt.Errorf("`version` is required when `create_mode` is %s", string(mongoclusters.CreateModeDefault))
+				}
+			case string(mongoclusters.CreateModeGeoReplica):
+				if state.SourceServerId == "" {
+					return fmt.Errorf("`source_server_id` is required when `create_mode` is `GeoReplica`")
+				}
+				if state.SourceLocation == "" {
+					return fmt.Errorf("`source_location` is required when `create_mode` is `GeoReplica`")
+				}
+			}
+
+			if state.ComputeTier == "Free" || state.ComputeTier == "M25" {
+				if state.HighAvailabilityMode == string(mongoclusters.HighAvailabilityModeZoneRedundantPreferred) {
+					return fmt.Errorf("high Availability is not available with the 'Free' or 'M25' Cluster Tier")
+				}
+
+				if state.ShardCount > 1 {
+					return fmt.Errorf("the value of `shard_count` cannot exceed 1 for the 'Free' or 'M25' Cluster Tier")
```

Comment:
The comments, log and error messages keep referring to a "Cluster Tier" but the property is called `compute_tier`, can we please try to be consistent and correct to avoid confusion
```suggestion
					return fmt.Errorf("high Availability is not available with the `Free` or `M25` Compute Tier")
				}

				if state.ShardCount > 1 {
					return fmt.Errorf("the value of `shard_count` cannot exceed 1 for the `Free` or `M25` Compute Tier")
```

Reply:
Fixed.